src/main/java/cz/klemsa/tkbrew/model/recipe/Recipe.java

```diff
@@ -0,0 +1,159 @@
+package cz.klemsa.tkbrew.model.recipe;
+
+import org.bson.types.ObjectId;
+import org.springframework.data.annotation.Id;
+import org.springframework.data.annotation.PersistenceConstructor;
+import org.springframework.data.mongodb.core.mapping.Document;
+
+/**
+ * Class that's represents recipe.
+ */
+@Document(collection = "recipe")
+public final class Recipe {
+    private static final String DEFAULT_NAME = "";
+    private static final String DEFAULT_DESCRIPTION = "";
+    private static final WaterTable DEFAULT_WATER_TABLE = new WaterTable();
+    private static final Mash DEFAULT_MASH = new Mash();
+    private static final Boil DEFAULT_BOIL = new Boil();
+    private static final Fermentation DEFAULT_FERMENTATION = new Fermentation();
+
+    @Id
+    private final ObjectId  objectId;
+
+    private final String name;
+    private final String description;
+    private final WaterTable waterTable;
+    private final Mash mash;
+    private final Boil boil;
+    private final Fermentation fermentation;
+
+    /**
+     * Recipe builder.
+     */
+    public static class RecipeBuilder {
```

Comment:
This is not Builder pattern.
See this: https://dzone.com/articles/design-patterns-the-builder-pattern
(and return "this"...)
Question is: Is builder pattern something you want to use for recipe? (implying that recipe could be incomplete before building) ?